scrapy/core/engine.py

```diff
@@ -184,7 +186,8 @@ def _next_request_from_scheduler(self) -> Optional[Deferred]:
         d.addErrback(lambda f: logger.info('Error while removing request from slot',
                                            exc_info=failure_to_exc_info(f),
                                            extra={'spider': self.spider}))
-        d.addBoth(lambda _: self.slot.nextcall.schedule())
+        slot = self.slot
+        d.addBoth(lambda _: slot.nextcall.schedule())
```

Comment:
This is also necessary for the same reason, because `close_spider` uses `setattr` to set `self.slot` to `None`. However, I wonder if we can handle this differently, or somehow prevent that `setattr` call to happen before this code is called.